### PR TITLE
Remove 'monet-pimp' from tests

### LINKED ISSUE #64: List `.map` does not always return a new List
This:

```typescript
List.fromArray(['a', 'b', 'c']).map(_ => null)
```

and this

```typescript
List.fromArray(['a', 'b', 'c']).map(_ => undefined)
```

returns `curry` function instead of a new List.

### PR DESCRIPTION
**Pull request by ulfryk**
Now we are pretty sure that `monet` will work well without `monet-pimp` -- no native prototypes extensions are added when pure `monet` is used.

This PR also fixes #64 and #79

### CODE REVIEW

File: dist/monet.js [hunk 1]
```diff
@@ -197,15 +206,17 @@
...
+            if (arguments.length === 0) {
                 this.isNil = true;
                 this.size_ = 0;
             } else {
                 this.isNil = false;
                 this.head_ = head;
-                this.tail_ = tail == null ? Nil : tail;
-                this.size_ = tail == null ? 0 : tail.size() + 1;
+                this.tail_ = tail || Nil;
+                this.size_ = this.tail_.size() + 1;
```
Comment (cwmyers) [inline dist/monet.js]: Awesome -- thanks for fixing this!

File: dist/monet.js [hunk 2]
```diff
@@ -824,7 +835,7 @@
         return Return(a);
     };
     Free.liftF = function(functor) {
-        return Suspend(functor.map(Return));
+        return isFunction(functor) ? Suspend(compose(Return, functor)) : Suspend(functor.map(Return));
```
Comment (cwmyers) [inline dist/monet.js]: I'm not completely sure about this one.  `liftF` lifts a functor into the Free monad.  Functions are functors, but I wonder if this should be a different method.  Could it lead to confusion?

File: dist/monet.js [hunk 3]
```diff
@@ -841,9 +852,9 @@
             });
         },
         bind: function(fn) {
-            return this.isSuspend ? Suspend(isFunction(this.functor) ? compose(function(free) {
+            return this.isSuspend ? isFunction(this.functor) ? Suspend(compose(function(free) {
```
Comment (cwmyers) [inline dist/monet.js]: Ah, was there a bug here?

File: dist/monet.js (hunk 2 above)
Comment (ulfryk) [inline dist/monet.js]: `Function` does not have `.map()` property in JS - so we have to treat it differently :(

File: dist/monet.min.js [hunk 4]
```diff
@@ -6,5 +6,5 @@
...
...z(isFunction(this.functor)?compose(function(n){return n.bind(t)},this.functor):this.functor.map(function(n){return n.bind(t)})):t(this.val)},ap:function(t){return this.bind(function(n){return t.map(function(t){return t(n)})})},resume:function(){return this.isSuspend?_(this.functor):I(this.val)},go1:function(t){function go2(n){return n.resume().cata(function(n){return go2(t(n))},idFunction)}return go2(this)},go:function(t){for(var n=this.resume();n.isLeft();){var i=t(n.left());n=i.resume()}return n.right()}},j.fn.init.prototype=j.fn,n.Identity=Identity,Identity.of=function(t){return new Identity(t)},Identity.fn=Identity.prototype={init:function(t){this.val=t},bind:function(t){return t(this.val)},get:function(){return this.val},equals:function(t){return isFunction(t.get)&&equals(this.get())(t.get())},toString:function(){return"Identity("+this.val+")"},inspect:function(){return""+this}},Identity.fn.init.prototype=Identity.fn,decorate(R),decorate(q),decorate(d),decorate(V),decorate(NEL),decorate(List),decorate(g),decorate(O),decorate(j),decorate(Identity),d.fromNull(t).filter(function(t){return!!t.notUseMonetGlobalObject}).cata(function(){return r(e,n)},function(t){return r(t,n),e})});
```
Comment (cwmyers) [inline dist/monet.min.js]: Do you feel we should check in generated code?

File: package.json [hunk 5]
```diff
@@ -30,8 +30,9 @@
   "scripts": {
     "test": "karma start && tsc -m system --noEmit --listFiles",
     "test-typings": "tsc -m system --noEmit --listFiles",
-    "tdd": "karma start --auto-watch  --no-single-run --browsers Chrome,Firefox,PhantomJS",
-    "tdd-osx": "karma start --auto-watch  --no-single-run --browsers Chrome,Firefox,PhantomJS,Safari",
+    "tdd": "karma start --auto-watch --no-single-run",
+    "tdd-full": "karma start --auto-watch --no-single-run --browsers Chrome,Firefox,PhantomJS",
+    "tdd-full-osx": "karma start --auto-watch --no-single-run --browsers Chrome,Firefox,PhantomJS,Safari",
```
Comment (cwmyers) [inline package.json]: 👍

File: dist/monet.js (hunk 3 above)
Comment (ulfryk) [inline dist/monet.js]: not just cleaning up code this time

File: dist/monet.js (hunk 2 above)
Comment (cwmyers) [inline dist/monet.js]: Good point.

File: dist/monet.min.js (hunk 4 above)
Comment (ulfryk) [inline dist/monet.min.js]: I'll drop it in future (like just before release of `0.9.0` stable ) -- it just helped me to push things faster during development

**Pull request merged**